HIP/hip-631.md

```diff
@@ -0,0 +1,555 @@
+
+#### Contracts
+
+1. Traverse all contract accounts
+2. Check `contract.alias`
+3. If `contract.alias` is a Hedera address or empty then skip
+4. If `contract.alias` is a 20-byte hex perform `CryptoUpdate` with `virtual_address_update.add.evm_address = account.alias` and `virtual_address_update.add.is_default = true`
+
+### Connect to Hedera via Virtual Address ECDSA key
+
+An account should be searchable by its virtual addresses fi applicable.
+
+Search by `evmAddress` will be supported by consulting the `evmAddress` → `AccountId` map on queries. Similarly, if an `ECDSA` public key is provided instead, the network should support extraction of the public address and subsequent search by the retrieved value. This will expand account import options in wallets and allow for account info retrieval in more scenarios.
+
+## Transaction signing with matching simple key
+
+There will be cases where users may have a simple key (`ECDSA` or `ED25519`) that does not correlate to the virtual address used in a transaction that requires `ECDSA` key signing. In this case users may want to utilize their virtual address key whiles maintaining the existing simple key(s) they have on their account.
+
+To support such mobility cases, where the security implications make sense a user may update their simple key to a 1 of n threshold key. In this way users can sign transactions with their initial simple key or their desired `ECDSA` simple key that maps to an evmAddress.
+
+For example an early adopter of Hedera will most certainly have an `ED25519` key on their account. In this way they would not be able to successfully call smart contract transactions that utilize `ECRECOVER`. With the addition of a virtual address they are 1 step closer to being able to support this scenario but still do not match the `ECDSA` key signing requirements. In this case the user may update their `ED25519` key to a 1 of 2 threshold key where transactions may be signed by either the `ED25519`  or `ECDSA` private key the user has control over.
+
+In this scenario it can be seen as a safe security modification, however this should be up to the discretion of the user.
+
+Note: This scenario is reserved for simple key cases only. It is strongly advised that users do not modify complex keys to be a 1 of n in which the complex key is on one of the options as this dilutes the original security architecture.
+
+## Contract account virtual address
+
+With the addition of virtual addresses and the separation of alias, network logic allows a contract account to represent its contract address with its virtual address.
+
+## Community product impacts
+
+### SDK Impacts
+
+To support these updates the SDK will need to update the available methods on some transactions
+
+1. `CryptoUpdate` transactions will need support to add/remove and specify default `evmAddress`.
```

Comment:
We should add disable as an option here as well.

Reply:
Added